.github/workflows/ui_tests.yml

```diff
@@ -326,11 +327,94 @@ jobs:
         if: failure()
         with:
           name: playwright-report-${{ matrix.shard }}
-          path: ./apps/laboratory/playwright-report/
+          path: ${{ github.workspace }}/apps/laboratory/playwright-report/
+          retention-days: 7
+
+      - name: Upload Playwright Blob Report for Shard ${{ matrix.shard }}
+        uses: actions/upload-artifact@v4
+        if: always() # Ensure it runs even if tests fail to get all blob data
+        with:
+          name: playwright-blob-artifact-shard-${{ matrix.shard }} # New unique artifact name pattern for blobs
+          path: ./apps/laboratory/playwright-blob-reports/report-${{ matrix.shard }}.zip # Path to the uniquely named blob file
           retention-days: 7
 
       - name: Verify Cache Dir Exists Before Post Job
         run: |
           echo "Checking cache directory just before post-job phase:"
           ls -la /tmp/playwright_cache # Use workspace path
         continue-on-error: true # Don't fail the job if ls fails
+
+  merge_reports:
+    needs: ui_tests # Ensures this job runs only after all 'test' (sharded) jobs complete
+    if: always() # Run this job even if some test shards failed, to get a complete picture
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - uses: pnpm/action-setup@v4
+        name: Install pnpm
+
+      - name: setup-node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22.x
+          cache: 'pnpm'
+
+      # Download actual artifact in GitHub Actions
+      - name: Download node_modules (GitHub Actions)
+        uses: actions/download-artifact@v4
+        with:
+          name: ui-node-modules
+          path: node_modules # Specify path to download into
+
+      # Restore executable permissions for binaries (keep this after potential copy)
+      - name: Fix permissions
+        run: chmod +x node_modules/*/bin/* || true # Allow failure if no binaries found
```

Comment:
Is there any security concern?

Reply:
I'll try to remove this entirely - let's see if it's needed